src/online2/online-nnet2-feature-pipeline.cc

```diff
@@ -149,15 +174,28 @@ void OnlineNnet2FeaturePipeline::GetAdaptationState(
   // else silently do nothing, as there is nothing to do.
 }
 
+void OnlineNnet2FeaturePipeline::SetCmvnState(
+    const OnlineCmvnState &cmvn_state) {
+  cmvn_feature_->SetState(cmvn_state);
+}
+
+void OnlineNnet2FeaturePipeline::GetCmvnState(
+    OnlineCmvnState *cmvn_state) {
+  int32 frame = cmvn_feature_->NumFramesReady() - 1;
+  // the following call will crash if no frames are ready.
+  cmvn_feature_->GetState(frame, cmvn_state);
+}
+
 
 OnlineNnet2FeaturePipeline::~OnlineNnet2FeaturePipeline() {
   // Note: the delete command only deletes pointers that are non-NULL.  Not all
   // of the pointers below will be non-NULL.
   // Some of the online-feature pointers are just copies of other pointers,
   // and we do have to avoid deleting them in those cases.
-  if (final_feature_ != feature_plus_optional_pitch_)
+  if (final_feature_ != feature_plus_optional_cmvn_)
```

Comment:
Probably some of the memory release code here should handle the NULL pointer case, like [here](https://github.com/kaldi-asr/kaldi/pull/3462/files#diff-297989e9a4e929019e7b6db4e1b549f4R214), e.g., `ivector_feature_`, `feature_plus_optional_cmvn_`, `pitch_feature_`, `pitch_`, etc